File: boot-cache/src/main/scala/com/hjf/boot/demo/boot/cache/scala/Hello.scala
package com.hjf.boot.demo.boot.cache.scala

/**
  * name: com.hjf.boot.demo.boot.cache.scala
  * author: mickjoust
  * date: 2018/5/14
  **/
class Hello (val firstName: String, val lastName: String){
  def this(firstName: String) { this(firstName, "")}
}
object Hello {
  def toUpper(str: String): String = {
    return str.toUpperCase()
  }
}

object Main extends App {
  val p1 = new Hello("mickjoust","huang")
  val p2 = new Hello("mickjoust")

  println(Hello.toUpper("mickjoust"))

}

File: boot-other/src/main/kotlin/com/hjf/boot/demo/boot/cache/kotlin/hello.kt
package com.hjf.boot.demo.boot.cache.kotlin

/**
 * name: com.hjf.boot.demo.boot.high_concurrency.kotlin
 * author: mickjoust
 * date: 2018/5/9
 **/
fun main(args: Array<String>){
    println("Hello World");
}

class Person(val firstname: String,val lastname: String) {
    constructor(name: String) : this(name, "")
    fun printDetails() = println("FirstName: ${firstname}, LastName: ${lastname}")
}

interface ReportSender
{
    fun generateReport() : String
    fun sendReport() {
        val report = generateReport()
        println("Report: "+ report)
    }
}

class HTMLReportSender : ReportSender
{
    override fun generateReport(): String = "<html><body>ReportData</body></html>"
}

//fun main(args: Array<String>)
//{
//    val rs = HTMLReportSender()
//    rs.sendReport()
//}